lib/Service/EndpointService.php

```diff
@@ -666,6 +688,130 @@ private function processSyncRule(Rule $rule, array $data): array
 		return $data;
 	}
 
+    /**
+     * Processes a file part creation rule.
+     *
+     * @param Rule $rule The rule to process.
+     * @param array $data The created object in array form.
+     * @param string|null $objectId The id of the resulting object.
+     *
+     * @return array The updated object data.
+     *
+     * @throws ContainerExceptionInterface
+     * @throws DoesNotExistException
+     * @throws GuzzleException
+     * @throws LoaderError
+     * @throws MultipleObjectsReturnedException
+     * @throws NotFoundExceptionInterface
+     * @throws SyntaxError
+     * @throws \OCA\OpenRegister\Exception\ValidationException
+     * @throws \OCP\Files\InvalidPathException
+     * @throws \OCP\Files\NotFoundException
+     */
+    private function processFilePartRule(Rule $rule, array $data, Endpoint $endpoint, ?string $objectId = null): array
+    {
+        if($objectId === null) {
+            throw new Exception('Filepart rules can only be applied after the object has been created');
+        }
+
+        $config = $rule->getConfiguration()['fileparts_create'];
```

Comment:
Can this array key fileparts_create be null? If so should it not be checked before used to prevent errors

Reply:
Added a check for this